Marlin/src/module/planner.cpp

```diff
@@ -2358,10 +2358,10 @@ bool Planner::_populate_block(block_t * const block, bool split_move,
                 4.85301876f, 6.77019119f, 9.50873947f, 13.4009094f, 18.9188652f,
                 26.7320709f, 37.7884521f, 53.4292908f, 75.5522461f,  0.0f };
 
-              const int16_t idx = (t == 0.0f) ? 0 : __builtin_clz(int16_t((1.0f - t) * jd_lut_tll)) - jd_lut_tll0;
+              const int16_t idx = (t == 0.0f) ? 0 : __builtin_clz(uint16_t((1.0f - t) * jd_lut_tll)) - jd_lut_tll0;
 
               float junction_theta = t * pgm_read_float(&jd_lut_k[idx]) + pgm_read_float(&jd_lut_b[idx]);
-              if (neg > 0) junction_theta = RADIANS(180) - junction_theta;
+              if (neg < 0) junction_theta = RADIANS(180) - junction_theta;
```

Comment:
I reversed this condition intentionally, because the code that was originally submitted did this:
```cpp
if (neg < 0) junction_theta = RADIANS(180) - junction_theta;
...
const float limit_sqr = (block->millimeters * junction_acceleration) / (RADIANS(180) - junction_theta);
```
By reversing the test, the output is `acos(-t)` and the subtraction is not needed in the divisor.

Reply:
I didn't notice this. Maybe we should put comment on that. Reverted.